From 45c73dfd98321dd1977cfa7b7cfb3a92c1fb9a29 Mon Sep 17 00:00:00 2001
From: Kohei Yoshida <kohei.yoshida@collabora.com>
Date: Mon, 30 Jun 2014 13:43:55 -0400
Subject: [PATCH 2/3] Don't go further and pop the stack if it's empty.

Change-Id: I27bd30ca65cf0066cd022b4b060757913ea01fed
(cherry picked from commit 3ede68e1cc879e6adbf6336eccaf88ced974b8b6)
Reviewed-on: https://gerrit.libreoffice.org/9989
Tested-by: Michael Stahl <mstahl@redhat.com>
Reviewed-by: Michael Stahl <mstahl@redhat.com>
---
 sax/source/fastparser/fastparser.cxx | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sax/source/fastparser/fastparser.cxx b/sax/source/fastparser/fastparser.cxx
index 577f414..4ddbd2a 100644
--- a/sax/source/fastparser/fastparser.cxx
+++ b/sax/source/fastparser/fastparser.cxx
@@ -492,6 +492,12 @@ void Entity::characters( const OUString& sChars )
 
 void Entity::endElement()
 {
+    if (maContextStack.empty())
+    {
+        // Malformed XML stream !?
+        return;
+    }
+
     const SaxContext& aContext = maContextStack.top();
     const Reference< XFastContextHandler >& xContext( aContext.mxContext );
     if( xContext.is() ) try
-- 
1.9.3

